crates/api/api/src/comment/like.rs

```diff
@@ -70,21 +72,33 @@ pub async fn like_comment(
     }
   }
 
-  let mut like_form = CommentLikeForm::new(local_user_view.person.id, data.comment_id, data.score);
+  let mut like_form = CommentLikeForm::new(my_person_id, data.comment_id, data.score);
 
   // Remove any likes first
-  let person_id = local_user_view.person.id;
+  CommentActions::remove_like(&mut context.pool(), my_person_id, comment_id).await?;
+  if let Some(previous_like_score) = orig_comment.comment_actions.and_then(|p| p.like_score) {
+    PersonActions::remove_like(
+      &mut context.pool(),
+      my_person_id,
+      orig_comment.creator.id,
+      previous_like_score,
+    )
+    .await
+    // Ignore errors, since a previous_like of zero throws an error
+    .ok();
+  }
 
-  CommentActions::remove_like(&mut context.pool(), person_id, comment_id).await?;
+  like_form = plugin_hook_before("before_comment_vote", like_form).await?;
+  let like = CommentActions::like(&mut context.pool(), &like_form).await?;
+  PersonActions::like(
+    &mut context.pool(),
+    my_person_id,
+    orig_comment.creator.id,
+    like_form.like_score,
+  )
+  .await?;
 
-  // Only add the like if the score isnt 0
-  let do_add =
-    like_form.like_score != 0 && (like_form.like_score == 1 || like_form.like_score == -1);
-  if do_add {
```

Comment:
I moved this validation check into the functions, it makes much more sense to do there anyway.